src/openrct2/world/Footpath.cpp

```diff
@@ -1396,26 +1392,27 @@ static int32_t footpath_is_connected_to_map_edge_recurse(
 
         do
         {
-            edges &= ~(1 << direction);
-            if (tileElement->AsPath()->IsSloped() && tileElement->AsPath()->GetSlopeDirection() == direction)
+            shiftedFootpathPos.direction = newDirection;
+            edges &= ~(1 << shiftedFootpathPos.direction);
+            if (tileElement->AsPath()->IsSloped() && tileElement->AsPath()->GetSlopeDirection() == shiftedFootpathPos.direction)
             {
-                z += 2;
+                shiftedFootpathPos.z += (2 * 8);
             }
-            int32_t result = footpath_is_connected_to_map_edge_recurse(x, y, z, direction, flags, level, 0, junctionTolerance);
+            int32_t result = footpath_is_connected_to_map_edge_recurse(shiftedFootpathPos, flags, level, 0, junctionTolerance);
             if (result == FOOTPATH_SEARCH_SUCCESS)
             {
                 return result;
             }
-        } while (get_next_direction(edges, &direction));
+        } while (get_next_direction(edges, &newDirection));
 
         return FOOTPATH_SEARCH_INCOMPLETE;
     }
 }
 
-int32_t footpath_is_connected_to_map_edge(int32_t x, int32_t y, int32_t z, int32_t direction, int32_t flags)
+int32_t footpath_is_connected_to_map_edge(const CoordsXYZD& footpathPos, int32_t flags)
 {
     flags |= (1 << 0);
```

Comment:
Future job will be to use the `GAME_COMMAND_FLAGS`

Reply:
I've added a `TODO` to track it.